src/modals/SendModals/hooks.tsx

```diff
@@ -0,0 +1,206 @@
+/*
```

Comment:
So, we've chosen another project structure philosophy, binding hooks, types etc. in the modal folder. 
For discoverability sake, I wonder if it wouldn't be better to go with something like for instance: 
`hooks/transactions/useBuildTxCommonComponents.ts` instead.

In short: centralising hooks, types etc. at the root of the project, and splitting them into unit files whenever possible. WDYT?

Reply:
IMO a hook should be usable from anywhere - we should ask ourselves, _should_ this be a hook?

I think keeping to `hooks/` was good organization but let's not get tied up on it...!

Reply:
> In short: centralising hooks, types etc. at the root of the project, and splitting them into unit files whenever possible. WDYT?

Yes, agreed.

> IMO a hook should be usable from anywhere

I disagree that that's the only use-case of a hook. IMO, a hook can also be used to remove complicated logic and keep a component as minimal as possible. With a good descriptive name, anyone going through the component code can understand what is going on, and if they want to learn about implementation details, they can have a look at the hook.

Reply:
Well... if we all agree, we should move the hooks and types to the respective directories at the root of the project 🤓

Reply:
Done! Merge? :pleading_face: :pray:

Reply:
No... I'm sorry, maybe it wasn't clear, but what we're proposing here is to group hooks (and types, as mentioned above) according to their functions, instead of associating them to specific UI components.

What you did here doesn't accomplish that goal as you simply took the hooks and put them in `hooks/modal/etc.`: they are still tied to some specific component.

What would make sense, as proposed above, is to group the hooks under `hooks/transactions/...` which would allow us to reuse them in some other places where some transaction related hooks are necessary. Same for the types: `types/transactions/...`.

I also suggested to split the hooks into dedicated files for discoverability sake. Here, you have duplicated `hooks` folders (`hooks/.../.../hooks`) which doesn't make sense.

I'm sorry, but I clearly put the emphasis on quality instead of speed. Let's not take shortcuts that we'll regret later.

Reply:
> It wasn't clear to me you wanted to split the file, i.e. 1 hook per file.

This isn't the main issue. The problem for me is that we want to group hooks according to their tasks, not tie them to specific UI component. 

Anyway, it's not a big deal, but I think it's important to state clearly that we don't want to cut any corner to save a few days. 🤝  That's how we build a solid company culture (and some solid code as well!) ❤️

Reply:
So as suspected the original code was not by definition a hook, and I further confirmed this with extensional examples, such as https://github.com/streamich/react-use. If we ever find ourselves transforming and manipulating components within hooks it should sound some alarm bells: this is a code smell.

The latest commit was able to completely remove the "hook".

Please let me know if there's anything else to adjust in the code, there was a lot which has changed. I need to test all 3 modals tomorrow morning.

Reply:
Thanks a lot! Let's aim at merging this on Monday, so that @nop33 can take a look as well.

Reply:
There are operational errors present that I'm working through with @polarker but that's not a blocker for code review, just for merging.

Reply:
I rebased and force pushed to fix conflicts.